entity_api/src/mutate.rs

```diff
@@ -59,19 +60,56 @@ impl UpdateMap {
         Self::default()
     }
 
+    /// Inserts a key-value pair into the map.
+    ///
+    /// If the key already exists, the value will be overwritten.
+    pub fn insert(&mut self, key: String, value: Option<Value>) {
+        self.map.insert(key, value);
+    }
+
     /// Retrieves a value from the map by its key.
     ///
     /// Returns an Option containing a reference to the Value if it exists,
     /// or None if the key is not found or the value is None.
-    pub fn get(&self, key: &str) -> Option<&Value> {
+    pub fn get_value(&self, key: &str) -> Option<&Value> {
         self.map.get(key).and_then(|opt| opt.as_ref())
     }
 
-    /// Inserts a key-value pair into the map.
-    ///
-    /// If the key already exists, the value will be overwritten.
-    pub fn insert(&mut self, key: String, value: Option<Value>) {
-        self.map.insert(key, value);
+    /// Removes a value from the update map and returns it.
+    /// Returns Error if the key doesn't exist or the value is not a valid string.
+    pub fn remove(&mut self, key: &str) -> Result<String, Error> {
```

Comment:
Somewhere down the line I'd like to think through this specific data structure a bit more. It's somewhat unwieldy. It may make sense to just use a `HashMap` directly.. not sure.

Reply:
I think it's less the data structure itself and more that the way it's implemented now, we need to explicitly wrap values in `Value` structs which exposes some seaORM constructs that I would prefer be hidden from this layer. We achieve semi decoupling by re-exporting the concepts through the app layers but maybe there's a better way. Not sure.

Reply:
I see a lot of Rust crates reexport from underlying crate types/methods, but I agree the strongest interfaces are those that normally try to avoid this. I wouldn't call this a high priority though as long as it's not feeling overly broken and ugly to work with today.